src/compiler/tsc.ts

```diff
@@ -227,7 +227,7 @@ namespace ts {
 
         if (commandLine.options.help) {
             printVersion();
-            printHelp();
+            printHelp(commandLine.options.all);

```

Comment:
Feels strange that you won't get help output if you only provide `--all`. Either check for it, give an error, or create a `--help-all`